internal/scorecard/alpha/scorecard.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alpha
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/version"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	Log = logrus.New()
+)
+
+type ScorecardFlags struct {
+	Config       string
+	Bundle       string
+	Selector     string
+	ListAll      bool
+	OutputFormat string
+	Kubeconfig   string
+}
+
+// RunTests executes the scorecard tests as configured
+func RunTests(flags ScorecardFlags) error {
+	scConfig, err := lookupConfig(flags)
+	if err != nil {
+		return err
+	}
+
+	selector, err := labels.Parse(flags.Selector)
+	if err != nil {
+		return err
+	}
```

Comment:
It might be nice to decouple the flags from what we pass into the scorecard execution library.

For example, on the `cmd` CLI side of things, we can load the config file, parse the label selector, construct the kubernetes client, decide whether to list or run, and format the output. All of these could be separate functions provided by the scorecard package, but called from the `cmd` side.

On this side the `Options` struct could be:

```go
type Options struct {
  Config Config
  Selector labels.Selector
  Client kubernetes.Interface
}
```

This would give us better separation of concerns. WDYT?

Reply:
agree, updated in latest commit.